index.js

```diff
@@ -1,68 +1,65 @@
+import { monsters } from './monsters.js';
 
-import {monsters} from './monsters.js';
-console.log(monsters);
-for(let monsterDiv of monsters){
-    showMonster(monsterDiv);
+const monsterContainer = document.querySelector('.monsters');
+const monsterFilter = document.querySelector('#search-monster');
 
+function populateMonsters(content) {
+  // Clear out existing content
+  monsterContainer.innerHTML = '';
+  // Usde showMonster to generate an Array of DOMNodes
+  let createMonsters = content.map((el) => showMonster(el));
+  // Append the DOMNodes in one DOM transaction
+  monsterContainer.append(...createMonsters);
 }
 
-function showMonster(monsterDiv){
-    const monster = document.createElement('div');
-    monster.className = 'monster';
-    const img= document.createElement('img');
-    img.src = `https://robohash.org/${monsterDiv.id}?set=set2`;
-    img.alt = monsterDiv.name ;
-    const mname = document.createElement('p');
-    mname.className = 'name' ;
-    mname.innerText = monsterDiv.name;
-    const email = document.createElement('p');
-    email.className = 'email'
-    email.innerText = monsterDiv.email;
-    monster.append(img,mname,email);
-    document.querySelector('.monsters').append(monster);
-    console.log(monster);
-}
+function showMonster(monsterDiv) {
```

Comment:
This function does not inject anything into the DOM now and it simply returns a DOM node for the data provided.